server/src/main/java/org/apache/uniffle/server/buffer/ShuffleBufferWithSkipList.java

```diff
@@ -111,8 +111,26 @@ public int getBlockCount() {
   }
 
   @Override
-  public void release() {
-    blocksMap.values().forEach(spb -> spb.getData().release());
+  public long release() {
```

Comment:
I think we need to add synchronized here. Calling the `toFlushEvent` method and the `release` method at the same time may result in duplicate release.

Reply:
`toFlushEvent` method do not release buf, the buf released by a thread start by `ProcessEventThread` thread, it guarded by `app read lock`, but not `ShuffleBufferWithSkipList`.

Reply:
> `toFlushEvent` method do not release buf, the buf released by a thread start by `ProcessEventThread` thread

I know this.


> it guarded by `app read lock`

I don't get this point.

And is it possible that the following case？

1.  Thread-1( toFlushEvent): List<ShufflePartitionedBlock> spBlocks = new LinkedList<>(blocks);
2. Thread-2(release): blocks.forEach(spb -> spb.getData().release());  (The first release)
3. Thread-1( toFlushEvent): blocks.clear();
4. Thread-3 (DefaultFlushEventHandler#dispatchEvent) (The second release)

Reply:
I see. But i think relying on app locks is not a reliable solution. The `release` method may be used by other logic  in the future, and the same problem may occur again. It's ok for me not to change it.

Reply:
I agree with your point.